File: repos_estudiantes_TARQUI_tercera_entrega/ROJAS_ALCON_VICTOR_ALEJANDRO/generecidad/ej1/ej1.py
from typing import Generic, TypeVar

T = TypeVar('T')

class Caja(Generic[T]):
    def _init_(self):
        self.contenido = None

    def guardar(self, item: T):
        self.contenido = item

    def obtener(self) -> T:
        return self.contenido

caja_texto = Caja[str]()
caja_texto.guardar("Hola Auxi")

caja_numero = Caja[int]()
caja_numero.guardar(123456)

print("Caja de texto:", caja_texto.obtener())
print("Caja de número:", caja_numero.obtener())

File: repos_estudiantes_TARQUI_tercera_entrega/ROJAS_ALCON_VICTOR_ALEJANDRO/generecidad/ej1/ej1.java
class Caja<T> {
    private T contenido;

    public void guardar(T item) {
        contenido = item;
    }

    public T obtener() {
        return contenido;
    }
}

public class Main {
    public static void main(String[] args) {
        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("Hola Auxi");

        Caja<Integer> cajaNumero = new Caja<>();
        cajaNumero.guardar(123456);

        System.out.println("Caja de texto: " + cajaTexto.obtener());
        System.out.println("Caja de número: " + cajaNumero.obtener());
    }
}